File: 1-Part/3-chapter/reverseString.py
def rev(theString):
    if len(theString) == 0 or len(theString) == 1:
        return theString
    else:
        head = theString[0]
        tail = theString[1:]
        return rev(tail) + head


print(rev('abcdefghijklmnop'))
print(rev('Hello world'))

File: 1-Part/3-chapter/reverseString.js
function rev(theString) {
    if (theString.length === 0 || theString.length === 1) {
        return theString
    } else {
        let head = theString[0]
        let tail = theString.substring(1, theString.length)
        return rev(tail) + head
    }
}

console.log(rev('abcdefghijklmnop'))
console.log(rev('Hello world'))